### PR TITLE
More options

### PR DESCRIPTION
**Pull request by Farenheith**
## Fast Object LookUp

One of the most practical uses you can have with this package is to give a list of objects, to get a TrieMap:

```ts
const trie = objectListToTrie(myList);
```

This trie indexes all the objects in the list by each word, letting you iterate over it to find just the objects you need.
The object is also fully serializable as long as the list contains only serializable objects too.
You can specify some options to minimize the size of your Trie, like this:

```ts
const trie = objectListToTrie(myList, {
  minSize: 4, //Will only include in the Trie words with 4 or more characters,
  forbiddenWords: ['something', 'denied'], //Will Ignore those words when filling up the Trie
});
```

Tries generated with that method are case insensitive and accent insensitive too, and any non-alphanumeric character is ignored.
Given that, the searches you do over that Trie will also follow the same rules.
If, for any reason, you need a case sensitive Trie, or a Trie that consider non-alphanumeric characters, you can specify so like this:

```ts
const trie = objectListToTrie(myList, {
  minSize: 4, //Will only include in the Trie words with 4 or more characters,
  forbiddenWords: ['something', 'denied'], //Will Ignore those words when filling up the Trie
  {
    caseInsensitive: false,
    onlyAlphaNumerics: false,
  }
});
```

### CODE REVIEW

File: src/get-transform-string.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { TrieOptions } from './types';
+
+const removeSpecialCharactersAndAccents = /([\u0300-\u036f]|[^0-9A-Za-z])/g;
+
+export function toInsensitiveString(s: string) {
+	s = s.toLowerCase();
```
Comment (DanielM08) [inline src/get-transform-string.ts]: Not necessary

File: src/iterate-trie-values.ts [hunk 2]
```diff
@@ -215,10 +229,10 @@ export function iterateTrieValues<TValue>(
 ): Iterable<IteratedTrieValue<TValue>>;
 export function* iterateTrieValues<TValue>(
 	trie: Trie<TValue>,
-	prefixOrOptions?: IteratingOptions<TValue> | string,
+	prefixOrPrefix?: IteratingOptions<TValue> | string,
```
Comment (DanielM08) [inline src/iterate-trie-values.ts]: The wrong name coming back :')

File: src/object-list-to-trie.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { addObject } from './add-object';
+
+export function objectListToTrie<TValue extends object>(
+	list: Iterable<TValue>,
+	synonyms?: string[][],
+	reservedWords?: TrieOptions,
+) {
+	const parameters = processCharSynonyms(synonyms || [], reservedWords);
+	const trie = createEmptyTrie(parameters);
+	trie.s![3] = 1;
```
Comment (DanielM08) [inline src/object-list-to-trie.ts]: What is the meaning of position '3'? A variable explaining the meaning before the assignment would be nice

**Review by DanielM08 — CHANGES_REQUESTED** [review]

File: src/object-list-to-trie.ts (hunk 3 above)
Comment (Farenheith) [reply src/object-list-to-trie.ts]: Position 1 indicates that the Trie is both accent and case insensitive.
You can create a Trie, though, and inconsistently add case-sensitive words to it, so this solution is not good yet.
I'm trying to figure out a better way to do it

Comment (Farenheith) [reply src/object-list-to-trie.ts]: @DanielM08 I removed this options confusing array position and let it all specified on the TrieOptions (checkout the updated PR description).

**Review by DanielM08 — APPROVED** [review]

**Pull request merged**